distribution/lib/Standard/Table/0.0.0-dev/src/Data/Table.enso

```diff
@@ -513,24 +513,22 @@ type Table
 
 
     ## Prototype Group By function
-    group_by : Column_Selector -> [Aggregate_Column] -> Problem_Behavior -> Warnings.Warning_System -> Table
-    group_by selector columns (on_problems=Report_Warning) (warnings=Warnings.default) =
+    aggregate : [Aggregate_Column] -> Problem_Behavior -> Warnings.Warning_System -> Table
+    aggregate columns (on_problems=Report_Warning) (warnings=Warnings.default) =
         # Grouping Key
-        key_columns = if selector.is_nothing then [] else
-            Table_Helpers.select_columns_helper internal_columns=this.columns selector=selector reorder=True on_problems=on_problems warnings=warnings
-        key_length = key_columns.length
-        make_key = if (key_length == 0) then _->(Group_By.key [1]) else i->(Group_By.key (key_columns.map v->(v.at i)))
+        map c = case c of
+            Aggregate_Column.Group_By _ _ -> True
+            _ -> False
+        key_columns = columns.filter map . map c->(c.resolve_column this c.column)
```

Comment:
This just looks weird, especially the `columns.filter map . map` part. I'd suggest using some different name for this helper function. It looks like a predicate so maybe `is_a_group_criterion` or `is_key` or something like that?